File: Client/src/Student/model/subject-student.ts

import { Identity } from "../../model/Identity";

export class SubjectStudent extends Identity{
   StudentId :number | null = null ;
   SubjectId : number | null = null;
   Score : number =0;
   ScoreLabel : string='';
   TeacherId :number | null = null;
   Trimestre : string ='';
   Year : string | null = '';
   Date : Date = new Date();



  static JsonInstance(entity: any): SubjectStudent {
   return entity ? Object.assign(new SubjectStudent(), entity ): new SubjectStudent();

  }

  static JsonInstaces(entity: any[]): SubjectStudent[] {
    return entity.map(item => SubjectStudent.JsonInstance(item));
  }

}

File: Api/EscuelaPrimaria/Entity/SubjectStudent.cs
namespace EscuelaPrimaria.Entity
{
    public partial class SubjectStudent :  Entity
    {
        public long Id { get; set; }
        public long StudentId { get; set; }
        public long TeacherId { get; set; }
        public long SubjectId { get; set; }
        public string? Trimestre { get; set; }
        public string? Year { get; set; }
        public bool? Active { get; set; }
        public decimal? Score { get; set; }
        public DateTime Date { get; set; }
        public virtual Student Student { get; set; }
        public virtual Subject Subject { get; set; }

    }
}
